src/sigmatcher/definitions.py

```diff
@@ -33,11 +52,36 @@ def check_strings(self, strings: List[str]) -> List[str]:
     def capture(self, value: str) -> List[str]:
         raise NotImplementedError()
 
+    @abstractmethod
+    def get_dependencies(self) -> List[str]:
+        raise NotImplementedError()
+
+    def resolve_macro(
+        self, results: Dict[str, Union[Result, Exception, None]], result_identifier: str, result_modifier: str
+    ) -> str:
+        result = results[result_identifier]
+        assert result is not None
+        assert not isinstance(result, Exception)
```

Comment:
This shouldn't be an assertion: it should be an error that is caught and logged in the top level function `analyze`

Reply:
This is an assert because the code should never arrive to this method if the result is an exception or None because of `Analyzer.check_dependencies()`

Reply:
:+1: